cmd/root/bundle.go

```diff
@@ -50,87 +50,100 @@ func getProfile(cmd *cobra.Command) (value string) {
 	return envlib.Get(cmd.Context(), "DATABRICKS_CONFIG_PROFILE")
 }
 
-// loadBundle loads the bundle configuration and applies default mutators.
-func loadBundle(cmd *cobra.Command, args []string, load func(ctx context.Context) (*bundle.Bundle, error)) (*bundle.Bundle, error) {
-	ctx := cmd.Context()
-	b, err := load(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	// No bundle is fine in case of `TryConfigureBundle`.
-	if b == nil {
-		return nil, nil
-	}
-
+// configureProfile applies the profile flag to the bundle.
+func configureProfile(cmd *cobra.Command, b *bundle.Bundle) diag.Diagnostics {
 	profile := getProfile(cmd)
-	if profile != "" {
-		diags := bundle.ApplyFunc(ctx, b, func(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
-			b.Config.Workspace.Profile = profile
-			return nil
-		})
-		if err := diags.Error(); err != nil {
-			return nil, err
-		}
-	}
-
-	diags := bundle.Apply(ctx, b, bundle.Seq(mutator.DefaultMutators()...))
-	if err := diags.Error(); err != nil {
-		return nil, err
-	}
-
-	return b, nil
-}
-
-// configureBundle loads the bundle configuration and configures it on the command's context.
-func configureBundle(cmd *cobra.Command, args []string, load func(ctx context.Context) (*bundle.Bundle, error)) error {
-	b, err := loadBundle(cmd, args, load)
-	if err != nil {
-		return err
+	if profile == "" {
+		return nil
 	}
 
-	// No bundle is fine in case of `TryConfigureBundle`.
-	if b == nil {
+	return bundle.ApplyFunc(cmd.Context(), b, func(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+		b.Config.Workspace.Profile = profile
```

Comment:
Note: This is a divergence from the existing behaviour where the `--profile` flag is ignored if a profile is specified in the bundle config. 
After this change the `--profile` flag value will be used. This should have been the behaviour in the first place so this is great though, just wanted to call this out. 

Can you also add a unit test to encode this behaviour and prevent a regression here. This could also be a breaking change for some customers and should be called out in the release notes.

Reply:
I'm unable to reproduce the behavior you describe. For me the `--profile` flag always takes precedence.

Regardless of this, I have added test coverage to confirm this behavior.